apps/HelloAndroidCamera2/src/com/example/helloandroidcamera2/Camera2BasicFragment.java

```diff
@@ -0,0 +1,568 @@
+                        CameraCharacteristics.SCALER_STREAM_CONFIGURATION_MAP);
+
+                // See if the camera supports a YUV resolution that we want and create an
+                // ImageReader if it does.
+                List<Size> sizes = Arrays.asList(map.getOutputSizes(
+                        ImageFormat.YUV_420_888));
+                Log.d(TAG, "Available YUV_420_888 sizes:");
+                for (Size size : sizes) {
+                    Log.d(TAG, size.toString());
+                }
+
+                // You can use chooseOptimalSize() here to find the smallest size that exceeds the
+                // desired one which also matches the desired aspect ratio. But then you will need
+                // to set the SurfaceView aspect ratio again in case it does not match, and also
+                // use JNI to reconfigure the buffers again to match the new ImageReader size.
+                if (sizes.contains(IMAGE_READER_SIZE)) {
+                    // Create an ImageReader to receive images at that resolution.
+                    mImageReader = ImageReader.newInstance(IMAGE_READER_SIZE.getWidth(),
+                            IMAGE_READER_SIZE.getHeight(), ImageFormat.YUV_420_888,
+                            IMAGE_READER_BUFFER_SIZE);
+                    mImageReader.setOnImageAvailableListener(
+                            mOnImageAvailableListener, mBackgroundHandler);
+
+                    // Save the camera id to open later.
+                    mCameraId = cameraId;
+                    return;
+                }
+            }
+        } catch (Exception e) {
+            new ErrorDialog().show(getFragmentManager(), "dialog");
+        }
+    }
+
+    /**
+     * Opens the camera specified by {@link Camera2BasicFragment#mCameraId}.
+     */
+    private void openCamera() {
+        Activity activity = getActivity();
+        CameraManager manager = (CameraManager) activity.getSystemService(Context.CAMERA_SERVICE);
+        BlockingCameraManager blockingManager = new BlockingCameraManager(manager);
+        try {
+            mDeviceCallback = new BlockingStateCallback() {
+
+                @Override
+                public void onDisconnected(CameraDevice camera) {
+                    camera.close();
+                    mCameraDevice = null;
+                }
+
+                @Override
+                public void onError(CameraDevice camera, int error) {
+                    camera.close();
+                    mCameraDevice = null;
+                    Activity activity = getActivity();
+                    if (activity != null) {
+                        activity.finish();
+                    }
+                }
+            };
+
+            mCameraDevice = blockingManager.openCamera(mCameraId, mDeviceCallback,
+                    mBackgroundHandler);
+            createCameraPreviewSession();
+        } catch (BlockingOpenException|TimeoutRuntimeException e) {
```

Comment:
I pulled this locally and built, and this failed with

    } catch (BlockingOpenException|TimeoutRuntimeException e) {
    error: multi-catch statement is not supported in -source 1.5